src/trace-span.js

```diff
@@ -17,13 +17,26 @@
 'use strict';
 
 var util = require('util');
+var constants = require('./constants.js');
+
+function truncate(string, length) {
```

Comment:
You do need a big comment here describing the rationale for this. Also, what are the semantics of this function, it gatuantees that the returns string will have no more than `length` bytes, or does it guarantee that it will have less than `length` bytes?